docs/client-side-architecture.md

```diff
@@ -0,0 +1,118 @@
+
+- a dedicated core to power konnectors as autonomous entities
+- a set of useful components to build konnectors views (username, password, directory-select, tokens…)
+- a default view for basic konnectors (username / password / directory-target)
+- keep each konnector separated from the core
+- a minimalist API to interact with core features (e.g. to submit / validate form)
+- an easy way to override default behaviors (logics, views, styles…) for a given konnector
+
+
+### Technical recommendation
+
+We suggest to build the new version of Konnector client-side app using _Vue.js_ with _vue-router_ and _vue-resource_ for the app logics, and Webpack for the packaging tool along the Vue-loader webpack's loader.
+
+Vue.js is a frontend reactive framework useful to build good shaped interfaces and is built with design in mind. It doesn't need any complicated source logics (such as immutables sources, stores, etc) and instead keep focused on how to design components efficiently. Vue-router is its attached routing library, which makes URL based Ui really simple to develop.
+
+The Components approch of Vue.js, and its native filtering capabilities makes it a good candidate to build the new Konnectors Ui. It lets separate:
+
+- Main collection views (wizard, categories, configured konnectors…)
+- Ui components (input fields, secured fields, directory-selectors, interval-timer…)
+- Information subview (what does this konnector do?)
+- Configuration subview, which can be a dedicated one of the default built-in version
+
+A clever approach furbished by Vue.js ecosystem is the Vue-loader webpack's loader: it allows to  wrap in a single file all the needs to a dedicated component: template (markup), styles, and JS logics. This will allow community developers that need to create a custom view for their konnectors to wrap in one place all their behaviors, which will simplify contribution and maintenance, and will bring much more flexibility when we want to add features.
+
+
+### Core API
+
+The core will provide to each konnector component a set of methods (a minimal API) which allow it to interact with the functionnalities it needs. Those methods can be invoked using the `$emit` call (plus args) from the konnector component:
+
+#### `save`
+
+Send the data for the model to be saved.
+
+
+#### `validate`
+
+Ask the model to validate given data. Returns errors via `props`.
+
+
+#### `reset`
+
+Force the model to reset its data to default.
+
+
+#### `delete`
+
+Remove the model from the configured konnectors collection.
+
+
+### Components
+
+The core should provide a set of components ready to be used by konnectors:
+
+- input field
+- username field
+- password field
+- directory selector
+- interval timer selector
+
+
+### Default konnector view
+
+The default konnector view, which can be used by any konnector which doesn't need any custom form will provide the following fields:
+
+- username
+- password
+- directory
+- timer
+
```

Comment:
IMO, tokens are an input field (maybe a password type? not sure at this time). If we develop a Konnector which need a token, we just have to create a dedicated Vue.js component that build a custom form, reusing the components offered by the core. It may be something like:

```vue
<template>
    <k-username/>
    <k-token />
    <k-interval/>
</template>
```

and it's done :)